packages/fx-core/src/plugins/resource/bot/appStudio/appStudio.ts

```diff
@@ -176,7 +186,7 @@ export class AppStudio {
             }
           },
           true
-        );
+        ).catch((e) => sendAppStudioApiTelemetry(telemetryReporter, e));
```

Comment:
Is it possible to do this catch and send telemetry outside the appStudio.ts? It looks too intrusive to the existing code to pass telemetryReport to each function call, which make code not clean. There is already a guideline in Bot plugin on how to handle exception and telemetry in a unified way, @IvanJobs can you help review this PR and suggest the right way to add telemetry for appStudio call?

Reply:
@XiaofuHuang I agree with @eriolchan on the intrusive part.
According to the original design, for example, https://github.com/OfficeDev/TeamsFx/blob/60d497a45bf7987c3037a0c06b131393ed644f37/packages/fx-core/src/plugins/resource/bot/index.ts#L100, all exceptions are supposed to be caught on upper layer - `index.ts` which will significantly simplify the logic, provide single source of truth and reduce a lot of scattered code like what it is in the PR.

How about moving the logic on top layer? We don't want `context` or `telemetryReporter` being passed for all lower layers functions like this.

Reply:
Thx. Moved it to telemetryReporter